### PR TITLE
EVG-6310 Add wrapper and tests for retrieving a user's patches

### PR DESCRIPTION
**Pull request by dominoweir**

### CODE REVIEW

File: src/evergreen.ts [hunk 1]
```diff
@@ -80,7 +94,19 @@ export class client {
...
+
+    /**
+     * Gets patches for a particular user
+     * 
+     * @param callback - function to process the response
+     * @param username - Evergreen user's username
+     * @returns nothing
+     */
+    public getPatches(callback: request.RequestCallback, username?: string) {
+        const url =  "json/patches/user/" + username;
```
Comment (john-m-liu) [inline src/evergreen.ts]: nit: I would name this "resource" since it's a path rather than a full URL

Comment (dominoweir) [reply src/evergreen.ts]: Makes sense, fixed

File: src/evergreen.ts [hunk 2]
```diff
@@ -1,15 +1,16 @@
...
 import * as request from "request";
 
 export class client {
     public username: string;
     public key: string;
     public serverURL: string;
+    public webURL: string;
 
-    constructor(username: string, key: string, serverURL: string) {
+    constructor(username: string, key: string, serverURL: string, webURL: string = "") {
```
[the remark this answers is not in the document]
Comment (dominoweir) [reply src/evergreen.ts]: Fixed, serverURL => apiURL and webURL => uiURL

File: src/evergreen.ts [hunk 3]
```diff
@@ -18,11 +19,17 @@ export class client {
      * @param callback - function to process the response
      * @param resource - resource to GET, can be a path
      * @param params - query params to append to the request URL, in the format {"param": "value"}
+     * @param useWeb - tells us whether to use the web or server URL defined in evergreen.yml
      * @returns nothing
      */
-    public getResource(callback: request.RequestCallback, resource: string, params?: object) {
-        const url = this.serverURL + "/" + resource + queryString(params);
-        request.get(this.formRequest(url), callback);
+    public getResource(callback: request.RequestCallback, resource: string, params?: object, useWeb: boolean = false) {
```
[the remark this answers is not in the document]
Comment (dominoweir) [reply src/evergreen.ts]: Fixed

File: src/evergreen.ts [hunk 4]
```diff
@@ -31,11 +38,17 @@ export class client {
      * @param callback - function to process the response
      * @param resource - resource to POST to, can be a path
      * @param body - body of the request, usually as an object
+     * @param useWeb - tells us whether to use the web or server URL defined in evergreen.yml
      * @returns nothing
      */
-    public postResource(callback: request.RequestCallback, resource: string, body: any) {
-        const url = this.serverURL + "/" + resource;
-        request.post(this.formRequest(url, body), callback);
+    public postResource(callback: request.RequestCallback, resource: string, body: any, useWeb: boolean = false) {
```
[the remark this answers is not in the document]
Comment (dominoweir) [reply src/evergreen.ts]: Fixed

File: src/evergreen.ts [hunk 5]
```diff
@@ -80,7 +108,18 @@ export class client {
...
+
+    /**
+     * Gets patches for a particular user
+     * @param callback - function to process the response
+     * @param username - Evergreen user's username
+     * @returns nothing
+     */
+    public getPatches(callback: request.RequestCallback, username?: string) {
+        const resource =  "json/patches/user/" + username;
+        this.getUIResource(callback, resource, {});
```
Comment (john-m-liu) [inline src/evergreen.ts]: "params" is marked as an optional parameter, so you don't need to pass an empty object

**Pull request merged**